### PR TITLE
rectangle and triangle

### PR DESCRIPTION
**Pull request by maxglassie**
There were some weird things with testing and the exporting. The pattern in the pre-work made a weird error when named in camelCase, Katie showed me a new pattern that worked.  

I am slow at JavaScript. The brackets are ugly.

### CODE REVIEW

File: rectangle.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+var area = function(rectangle) {
+  return rectangle.width * rectangle.height
+};
+
+var perimeter = function(rectangle) {
+  return (rectangle.width * 2) + (rectangle.height * 2)
+};
+
+var diagonal = function(rectangle) {
+  var width_squared = (rectangle.width * rectangle.width);
```
Comment (case-eee) [inline rectangle.js]: be sure to use camel case here to follow conventions when writing JS 👍

File: test/triangle_test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const assert = require('chai').assert
+const isTriangle = require('../triangle').isTriangle;
+
+describe("isTriangle", function(){
```
Comment (case-eee) [inline test/triangle_test.js]: tests are looking good 👍

File: triangle.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+
+var isTriangle = function(a, b, c) {
+  if ((a <= 0) || (b <= 0) || (c <= 0)) {
```
Comment (case-eee) [inline triangle.js]: there's a lot of conditionals here - any way we could DRY this up? or make it easier to read possibly? could we be more explicit with our variable names?

File: rectangle.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  var height_squared = (rectangle.height * rectangle.height);
+  return Math.sqrt(width_squared + height_squared);
+};
+
+var diagonal = function(rectangle) {
```
Comment (case-eee) [inline rectangle.js]: i'm not sure i'm seeing the difference between this and the function you defined on line 11. is there anything different between the two?

**Comment by maxglassie** [comment]
Thanks for your comments @case-eee. I'm making slow progress... I'll keep chipping away but I'm having trouble being productive with this cold. Learning a lot tho!

File: averages.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+var mean = Array.prototype.mean = function() {
+  var array = this;
+  length = array.length
+  sum = array.reduce(function (a, b) {
+    return a + b;
+  }, 0)
+  return sum / length
+}
+
+var median = Array.prototype.median = function() {
```
Comment (case-eee) [inline averages.js]: we probably don't want to add these functions (total, mean, and median) to the array's prototype. generally, we don't want to add things to JavaScript's built-in objects. we just want to create stand-alone functions and use them accordingly.

File: averages.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  var sorted = array.sort(function(a,b) {
+    return a - b;
+  });
+  var length = array.length
+  var medianIndex = Math.floor(length / 2)
+
+  return sorted[medianIndex]
+}
+
+exports.total = total;
```
Comment (case-eee) [inline averages.js]: the content of these functions looks solid though 👍

File: test/average_test.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+const assert = require('chai').assert
+const total = require('../averages').total;
+const mean = require('../averages').mean;
+const median = require('../averages').median;
+
+describe("array object returns the total", function(){
```
Comment (case-eee) [inline test/average_test.js]: 👍

File: pad.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+pry = require('pryjs')
+
+var pad = Array.prototype.pad = function(min, pad) {
```
Comment (case-eee) [inline pad.js]: we don't want to add things to the Array prototype, we just want to create stand-alone functions that handle a specific piece of logic

File: pad.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+  var array = this;
+
+  if (array.length <= min) {
+    return array
+  } else {
+    array.fill(pad)
+  }
+}
+
+exports.pad = pad;
```
Comment (case-eee) [inline pad.js]: pad isn't quite working yet. the return value isn't what is specified in the requirements. keep working on this one 👍

File: averages.js (hunk 5 above)
Comment (case-eee) [inline averages.js]: i take this back!!! the challenge asks you to do this (whooooops on my part!) - this is great 👍

**Comment by maxglassie** [comment]
completes factorial, countBetweenNumbers, and longestString

**Comment by case-eee** [comment]
looks great @maxglassie 🎉 keep up the hard work!

**Pull request closed without merging**